cmd/queue.go

```diff
@@ -0,0 +1,309 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"log"
+	"time"
+
+	"github.com/frain-dev/convoy/config"
+	redisqueue "github.com/frain-dev/convoy/queue/redis"
+	"github.com/frain-dev/convoy/util"
+	"github.com/spf13/cobra"
+)
+
+func addQueueCommand(a *app) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "queue",
+		Short: "Get info about queue",
+	}
+
+	cmd.AddCommand(getQueueLength(a))
+	cmd.AddCommand(getZSETLength(a))
+	cmd.AddCommand(getStreamInfo(a))
+	cmd.AddCommand(getConsumersInfo(a))
+	cmd.AddCommand(getPendingInfo(a))
+	cmd.AddCommand(checkEventDeliveryinStream(a))
+	cmd.AddCommand(checkEventDeliveryinZSET(a))
+	cmd.AddCommand(checkEventDeliveryinPending(a))
+	return cmd
+}
+
+//Get queue length, number of entries in the stream
+func getQueueLength(a *app) *cobra.Command {
+	var timeInterval int
+	cmd := &cobra.Command{
+		Use:   "length",
+		Short: "queue length",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			q := a.eventQueue
+			ctx := context.Background()
+			ticker := time.NewTicker(time.Duration(timeInterval) * time.Millisecond)
+
+			for {
+				select {
+				case <-ticker.C:
+					length, err := q.Consumer().Queue().Len()
+					if err != nil {
+						log.Printf("Error getting queue length: %v", err)
+					}
+					log.Printf("Queue Length: %+v\n", length)
+				case <-ctx.Done():
+					return nil
+				}
+			}
+		},
+	}
+	cmd.Flags().IntVar(&timeInterval, "interval", 2000, "Log time interval")
+	return cmd
+}
```

Comment:
I think it'll be helpful to get the queue length in terms of "capacity". This function might return 200, but I won't understand what the number means, does 200 mean it's full? Or does it mean it's half full. I'm only suggesting this if it's possible.